src/extension/alterschema/linter/enum_validation_keywords_default.h

```diff
@@ -0,0 +1,79 @@
+class EnumValidationKeywordsDefault final : public SchemaTransformRule {
+public:
+  EnumValidationKeywordsDefault()
+      : SchemaTransformRule{
+            "enum_validation_keywords_default",
+            "Setting validation keywords alongside `enum` is considered an "
+            "anti-pattern, as the enumeration choices already imply their "
+            "respective constraints"} {};
+
+  [[nodiscard]] auto
+  condition(const sourcemeta::core::JSON &schema,
+            const sourcemeta::core::JSON &,
+            const sourcemeta::core::Vocabularies &vocabularies,
+            const sourcemeta::core::SchemaFrame &,
+            const sourcemeta::core::SchemaFrame::Location &,
+            const sourcemeta::core::SchemaWalker &walker,
+            const sourcemeta::core::SchemaResolver &) const
+      -> sourcemeta::core::SchemaTransformRule::Result override {
+    if (!contains_any(vocabularies,
+                      {"https://json-schema.org/draft/2020-12/vocab/validation",
+                       "https://json-schema.org/draft/2019-09/vocab/validation",
+                       "http://json-schema.org/draft-07/schema#",
+                       "http://json-schema.org/draft-06/schema#",
+                       "http://json-schema.org/draft-04/schema#",
+                       "http://json-schema.org/draft-03/schema#",
+                       "http://json-schema.org/draft-02/schema#",
+                       "http://json-schema.org/draft-02/hyper-schema#",
+                       "http://json-schema.org/draft-01/schema#",
+                       "http://json-schema.org/draft-01/hyper-schema#",
+                       "http://json-schema.org/draft-00/schema#",
+                       "http://json-schema.org/draft-00/hyper-schema#"}) ||
+        !schema.is_object() || !schema.defines("enum") ||
+        !schema.at("enum").is_array() || schema.defines("$ref")) {
+      return false;
+    }
+
+    std::set<sourcemeta::core::JSON::Type> enum_types;
+    for (const auto &value : schema.at("enum").as_array()) {
+      enum_types.emplace(value.type());
+    }
+
+    if (enum_types.empty()) {
+      return false;
+    }
+
+    this->blacklist.clear();
+    for (const auto &entry : schema.as_object()) {
+      const auto metadata = walker(entry.first, vocabularies);
+      if (metadata.type == sourcemeta::core::SchemaKeywordType::Other ||
```

Comment:
Do we really need this `if`? What example would be problematic if we removed this?